### DISCUSSION TITLE
How do I find an element with a unique attribute

### DISCUSSION THREAD

**Discussion by chenqinggang001** [seleniumbase/SeleniumBase#1427] [Q&A] [answered]
As shown in figure：

This property is unique in the element, and it's dynamic (with two states), and I can't locate it with xpath or CSS selecter，How do you locate this element by attribute name

![image](https://user-images.githubusercontent.com/74290987/180239609-31bbfb91-aa6d-4e83-9bf9-3d717a8920e1.png)

![image](https://user-images.githubusercontent.com/74290987/180240175-fb09be41-5ade-44f4-ae72-0afcc6d42ce3.png)

**Comment by chenqinggang001** [comment]
I can't predict whether it's in a top or bottom state，So I want to locate it by attribute name

**Comment by mdmintz** [comment, marked as answer]
Selector by attribute: `'div[x-placement]'`
Selector by attribute and value: `'div[x-placement="top-start"]'`
Other selector by attribute and value: `'div[x-placement="bottom-start"]'`

**Comment by chenqinggang001** [comment]
Maybe it's my syntax, I tried all these ways, but I couldn't locate this element

```
self.click("x-placement", by='Selector by attribute', timeout=10, delay=1)
self.click("div[x-placement]", by='Selector by attribute', timeout=10, delay=1)
self.click("x-placement", by='attribute', timeout=10, delay=1)
self.click("div[x-placement]", by='attribute', timeout=10, delay=1)
```

**Reply by mdmintz** [reply]
Use ``self.click("div[x-placement]")``
(It's a CSS Selector, which is the default)

**Reply by chenqinggang001** [reply]
Thank you very much!
